src/overlay/soil.py

```diff
@@ -109,16 +109,16 @@ def water(self, pos, play_sound: bool = True):
 
     def plant(self, pos, seed, inventory):
         tile = self.tiles.get(pos)
-        seed_amount = inventory.get(seed)
         seed_type = SeedType.from_farming_tool(seed)
+        seed_amount = inventory.get(seed_type.as_ir())
```

Comment:
As-is, if that operation returns None, the result value is invalid. If the default returned value was 0, you would avoid an unwanted error if the NPCs tried to plant something